APISpec.md

```diff
@@ -0,0 +1,508 @@
+# Todo API Spec
+
+## API Resources
+
+* **User** - A User represents an owner of Todos and corresponds one-to-many to a Todo.
+
+* **Todo** - A Todo represents a task to be completed and corresponds many-to-one to a User.
+
+* **Session** - A Session represents an authenticated user session and corresponds many-to-one to a User.
+
+## Data Tables
+
+### Users:
+```
+ID - Guid (PK)
+Username - String
+Password - String (Hash)
+```
+
+### Todos:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Title - String
+Status - Enumified String
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+### Tokens:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Token - String
+Expiration - Datetime
+```
+
+## Data Models
+
+### User
+```
+ID - Guid
+Username - String
+Password - String (Hash)
+Todos - Todo[] (List of Todos for the user)
+Sessions - Session[] (List of active sessions for the user)
+```
+
+### Todo
+```
+ID - Guid
+User - User
+Title - String
+Status - Status
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+### Session
+```
+ID - Guid
+Token - String (JWT from auth0)
+User - User
+Expiration - Datetime
+```
+
+### Status
+```
+OPEN
+IN_PROGRESS
+COMPLETE
+```
+
+## Endpoints
+
+### Create New Todo
+---
+
+```
+POST /todos
+```
+
+Headers:
+```
+"Authorization": Bearer <Token>
+```
+
+Body: 
+```
+{
+    "meta": {},
+    "data": {
+        "title": "Buy groceries",
+        "status": "OPEN|IN_PROGRESS|COMPLETE"
+        "dueDate": "2022-04-01T13:02:37.000Z"
+    }
+}
+```
+
+**Response** - `201 Created`
+
+Headers: 
+```
+"Location": "/todos/{id}"
+```
+
+**Error Codes**:
+
+`400 Bad Request` - Request payload invalid OR dueDate provided not a valid date
+
+`401 Unauthorized` - Request does not have valid authentication to create todo
```

Comment:
Rather `"User/Client is not authenticated"`

Reply:
Updated